web-comp.js

```diff
@@ -0,0 +1,204 @@
+        price: 139.99
+    },
+];
+
+function getCatalogProductTemplate(productInfo){
+    const templateProduct = document.createElement("template");
+
+    templateProduct.innerHTML = `
+        <style>
+
+        .catalog-product{
+            background-color: white;
+            border: 1px solid black;
+            display: flex;
+            flex-direction: column;
+            flex-grow: 1;
+            max-width: 210px;
+            padding: 24px;
+        }
+
+        .squared-image-container{
+            aspect-ratio: 1;
+            border: 1px solid black;
+            width: 100%;
+        }
+
+        .product-image{
+            aspect-ratio: 1;
+            display: block;
+            max-height: 100%;
+            max-width: 100%;
+        }
+
+        .product-title{
+            margin: 0 auto;
+        }
+
+        .description-container{
+            border: 1px solid black;
+            height: 80px;
+            margin: 18px 0;
+            overflow: auto;
+            padding: 8px;
+        }
+
+        .product-price{
+            margin: 0 auto;
+            margin-bottom: 24px;
+        }
+
+        .product-actions{
+            display: flex;
+            justify-content: space-between;
+        }
+
+        .product-quantity-container{
+            align-items: center;
+            display: flex;
+            flex-wrap: nowrap;
+        }
+
+        .product-quantity{
+            margin: 0 16px;
+        }
+
+        </style>
+        <div class="products-container">
+            <div class="catalog-product">
+
+                <div class="squared-image-container">
+                    <img class="product-image" src="${productInfo.imageURL}" alt="${productInfo.altAlert}">
+                </div>
+
+                <h5 class="product-title">
+                    ${productInfo.name}
+                </h5>
+
+                <div class="description-container">
+                    ${productInfo.description}
+                </div>
+
+                <h5 class="product-price">
+                    ${productInfo.price}€
+                </h5>
+                    
+                <div class="product-actions">
+                    <span class="product-quantity-container">
+                        <button class="decrement-quantity-button">-</button>
+                        <span class="product-quantity">1</span>
+                        <button class="increment-quantity-button">+</button>
+                    </span>
+                </div>
+            </div>
+        </div>`            
+
+        return templateProduct;
+}
+
+class CatalogProduct extends HTMLElement {
+    constructor (catalogProductInfo){
+        super()
+        const shadow = this.attachShadow({ mode: "open"});
+
+        const productTemplateElement = getCatalogProductTemplate(catalogProductInfo);
+
+        shadow.appendChild(productTemplateElement.content.cloneNode(true));
+        this.title = shadow.querySelector(".container");
```

Comment:
Always review adviced/copy-pasted code especially if it's not your.

Use 2 word names

Reply:
the line was a dead code, i removed it.
fixed